Error returned when repo path is more than two levels deep

### Describe the bug

Scanner fails to pull the image when the image path has more than two elements.  E.g. `library/ubi8/python-39`.  This is a common configuration for Harbor proxy-cache repositories, though by no means unique to this scenario.  This issue occurs for both normal and proxy-cache repositories.

Full log output returned by Harbor:
```
2023-11-10T23:38:17Z [INFO] [/pkg/scan/job.go:387]: {
  "uuid": "5dbac9a9-7f73-11ee-b2f5-5a0015c37d65",
  "name": "Carbon Black",
  "description": "",
  "url": "http://cb-harbor-adapter.carbonblack.svc:8080",
  "disabled": false,
  "is_default": true,
  "health": "healthy",
  "auth": "",
  "access_credential": "[HIDDEN]",
  "skip_certVerify": false,
  "use_internal_addr": true,
  "adapter": "Carbon-Black",
  "vendor": "VMware",
  "version": "1.0",
  "create_time": "2023-11-10T02:46:36.555415Z",
  "update_time": "2023-11-10T02:53:36.90597Z"
}
2023-11-10T23:38:17Z [INFO] [/pkg/scan/job.go:387]: {
  "registry": {
    "url": "http://harbor-dev-core:80",
    "authorization": "[HIDDEN]"
  },
  "artifact": {
    "namespace_id": 3,
    "repository": "test/ubi8/python-39",
    "tag": "latest",
    "digest": "sha256:11d0b758d1d8b26e673892bd6651426aa4a52853f331263e9fb2e017a18b5517",
    "mime_type": "application/vnd.docker.distribution.manifest.v2+json"
  }
}
2023-11-10T23:38:17Z [INFO] [/pkg/scan/job.go:167]: Report mime types: [application/vnd.scanner.adapter.vuln.report.harbor+json; version=1.0 application/vnd.security.vulnerability.report; version=1.1]
2023-11-10T23:38:17Z [ERROR] [/pkg/scan/job.go:370]: scan job: submit scan request: v1 client: submit scan: Scan failed: Expected exactly 2 substrings but found 3: general response handler: unexpected status code: 500, expected: 202
```

Log output from the harbor-adapter (log_level: error) pod doesn't really have anything useful:

```
[GIN] 2023/11/11 - 00:53:04 | 200 |      51.051µs |     172.20.1.51 | GET      "/api/v1/metadata"
[GIN] 2023/11/11 - 00:53:16 | 500 |       80.38µs |     172.20.2.76 | POST     "/api/v1/scan"
```

### Reproduction steps

Steps assume the following:
- CB harbor-adapter has been deployed in a pod on the same cluster (but not necessarily the same namespace) as Harbor.
- CB harbor-adapter has been added to Harbor as an interrogation source and set as default.

1. Pull `registry.access.redhat.com/ubi8/python-39:latest`.
2. Re-tag the image for Harbor.  e.g. `docker tag registry.access.redhat.com/ubi8/python-39:latest harbor.example.com/library/ubi8/python-39:latest`.
3. Push the image to Harbor.
4. Trigger a manual scan in the `library` project.

### Expected behavior

I'm just now testing this product, but my assumption is that the scan data returned will be something like the builtin Trivy scanner.  At least a successful HTTP code.

### Additional context

I believe the bug to be here:
https://github.com/vmware/carbon-black-adapter-for-harbor/blob/ea2ec76178df2829f5e268df08645caf24eb74de/internal/adapter/imagescanning/adapter.go#L89-L93

--- a/internal/adapter/imagescanning/adapter.go
+++ b/internal/adapter/imagescanning/adapter.go
@@ -86,12 +86,6 @@ func (a Adapter) Scan(payload harbor.ScanRequest) (harbor.ScanResponse, error) {
 		return resp, err
 	}
 
-	splits := strings.Split(payload.Artifact.Repository, "/")
-	if len(splits) != 2 {
-		err := fmt.Errorf("Expected exactly 2 substrings but found %d", len(splits))
-		return resp, err
-	}
-
 	tag := payload.Artifact.Tag
 	if payload.Artifact.Tag == "" {
 		tag = payload.Artifact.Digest[7:]

--- a/internal/queue/queue.go
+++ b/internal/queue/queue.go
@@ -32,11 +32,10 @@ func Queue() chan string {
 }
 
 func Publish(imageInfo ImageInfo) string {
-	locker.Lock()
-	defer locker.Unlock()
-
 	scanID := uuid.New().String()
+	locker.Lock()
 	imageInfoMap[scanID] = &imageInfo
+	locker.Unlock()
 	queue <- scanID
 	return scanID
 }

--- a/internal/queue/queue.go
+++ b/internal/queue/queue.go
@@ -7,6 +7,7 @@ package queue
 
 import (
 	"github.com/google/uuid"
+	"sync"
 )
 
 var (
@@ -14,9 +15,14 @@ var (
 	queue chan string
 	// imageInfoMap will store all the information for the images about to scan
 	imageInfoMap map[string]*ImageInfo
+
+	locker sync.Mutex
 )
 
 func InitQueue(bufferSize int) {
+	locker.Lock()
+	defer locker.Unlock()
+
 	queue = make(chan string, bufferSize)
 	imageInfoMap = make(map[string]*ImageInfo)
 }
@@ -26,6 +32,9 @@ func Queue() chan string {
 }
 
 func Publish(imageInfo ImageInfo) string {
+	locker.Lock()
+	defer locker.Unlock()
+
 	scanID := uuid.New().String()
 	imageInfoMap[scanID] = &imageInfo
 	queue <- scanID
@@ -33,10 +42,16 @@ func Publish(imageInfo ImageInfo) string {
 }
 
 func Fetch(scanID string) (*ImageInfo, bool) {
+	locker.Lock()
+	defer locker.Unlock()
+
 	v, ok := imageInfoMap[scanID]
 	return v, ok
 }
 
 func Remove(scanID string) {
+	locker.Lock()
+	defer locker.Unlock()
+
 	delete(imageInfoMap, scanID)
 }

